packages/smooth_app/lib/pages/offline_data_page.dart

```diff
@@ -0,0 +1,241 @@
+}
+
+/// Updates the product in the localdatabse and returns the total number of products updated
+Future<int> updateLocalDatabaseFromServer(BuildContext context) async {
+  final LocalDatabase localDatabase = context.read<LocalDatabase>();
+  final DaoProduct daoProduct = DaoProduct(localDatabase);
+// We seperate the products into two lists, one for products that have a knowledge panel
+// and one for products that don't have a knowledge panel
+  final List<String> barcodes = await daoProduct.getAllKeys();
+  final List<String> productsWithoutKnowledgePanel = <String>[];
+  final List<String> completeProducts = <String>[];
+  for (int i = 0; i < barcodes.length; i++) {
+    final Product? productFromDb = await daoProduct.get(barcodes[i]);
+    if (productFromDb != null && productFromDb.knowledgePanels == null) {
+      productsWithoutKnowledgePanel.add(barcodes[i]);
+    } else {
+      completeProducts.add(barcodes[i]);
+    }
+  }
+  final List<ProductField> fieldsForCompleteProducts = ProductQuery.fields;
+  final List<ProductField> fieldsForProductsWithoutKnowledgePanel =
+      ProductQuery.fields;
+  fieldsForProductsWithoutKnowledgePanel.remove(ProductField.KNOWLEDGE_PANELS);
+  int totalUpdatedProducts = 0;
+// Config for the products that don't have a knowledge panel
+  final ProductSearchQueryConfiguration productSearchQueryConfiguration =
+      ProductSearchQueryConfiguration(
+    language: ProductQuery.getLanguage(),
+    country: ProductQuery.getCountry(),
+    fields: fieldsForProductsWithoutKnowledgePanel,
+    parametersList: <Parameter>[
+      BarcodeParameter.list(productsWithoutKnowledgePanel),
+    ],
+  );
+
+  final SearchResult result = await OpenFoodAPIClient.searchProducts(
+    ProductQuery.getUser(),
+    productSearchQueryConfiguration,
+  );
+  if (result.products != null) {
+    daoProduct.putAll(result.products!);
+    totalUpdatedProducts += result.products!.length;
+  }
+
+// Config for the complete products ie. products that have a knowledge panel
+  final ProductSearchQueryConfiguration
+      productSearchQueryConfigurationForFullProducts =
+      ProductSearchQueryConfiguration(
+    language: ProductQuery.getLanguage(),
+    country: ProductQuery.getCountry(),
+    fields: fieldsForCompleteProducts,
+    parametersList: <Parameter>[
+      BarcodeParameter.list(completeProducts),
+    ],
+  );
+
+  final SearchResult resultForFullProducts =
+      await OpenFoodAPIClient.searchProducts(
+    ProductQuery.getUser(),
+    productSearchQueryConfigurationForFullProducts,
+  );
+  if (resultForFullProducts.products != null) {
+    daoProduct.putAll(resultForFullProducts.products!);
+    totalUpdatedProducts += resultForFullProducts.products!.length;
+  }
+  return totalUpdatedProducts;
+}
+
+class _OfflineDataPageState extends State<OfflineDataPage> {
+  @override
+  Widget build(BuildContext context) {
+    const String headerAsset = 'assets/preferences/main.svg';
```

Comment:
We could add a own svg for offline mode, nothing to block though, you can add a todo comment if you want

Reply:
Yeah added a todo there